lib/task_listener.js

```diff
@@ -110,21 +108,23 @@ export default class TaskListener extends EventEmitter {
     return cancelListener;
   }
 
+  get availableCapacity() {
+    return Math.max(this.capacity - this.runningTasks.length, 0);
```

Comment:
When I say can happen, I mean it can happen intentionally... So we should obviously not log a warning.... But we might leave a not that this can happen in case of dynamic capacity change.. .Which is used as shutdown... I suspect dynamic capacity change could be used for other things later too... Like test devices that we talked about.